lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -205,48 +276,204 @@
             }
         });
 
-        AccountSettingsFieldViews.DropdownFieldView = AccountSettingsFieldViews.FieldView.extend({
+        AccountSettingsFieldViews.DropdownFieldView = AccountSettingsFieldViews.EditableFieldView.extend({
 
             fieldType: 'dropdown',
 
             templateSelector: '#field_dropdown-tpl',
 
             events: {
-                'change select': 'saveValue'
+                'click': 'startEditing',
+                'change select': 'finishEditing',
+                'focusout select': 'finishEditing'
             },
 
             initialize: function (options) {
+                _.bindAll(this, 'render', 'optionForValue', 'fieldValue', 'displayValue', 'updateValueInField', 'saveValue');
                 this._super(options);
-                _.bindAll(this, 'render', 'fieldValue', 'updateValueInField', 'saveValue');
+
                 this.listenTo(this.model, "change:" + this.options.valueAttribute, this.updateValueInField);
             },
 
             render: function () {
                 this.$el.html(this.template({
                     id: this.options.valueAttribute,
+                    mode: this.mode,
                     title: this.options.title,
+                    iconName: this.options.iconName,
                     required: this.options.required,
                     selectOptions: this.options.options,
                     message: this.helpMessage
                 }));
+
                 this.updateValueInField();
+
+                if (this.editable === 'toggle') {
+                    this.showCanEditMessage(this.mode === 'display');
+                }
                 return this;
             },
 
+            modelValueIsSet: function() {
+                return (this.modelValue() && this.optionForValue(this.modelValue())) == true;
```

Comment:
@muzaffaryousaf I think this should resolve the issue where dropdowns with values are showing the wrong hover styling.
```
var value = this.modelValue();
if (_.isUndefined(value) || _.isNull(value) || value == '') {
    return false;
} else {
    return !(_.isUndefined(this.optionForValue(value)))
}
```